### DISCUSSION TITLE
What are the current best practices for signing AIR applications for Windows?

### DISCUSSION THREAD

**Discussion by KaminoRyo** [airsdk/Adobe-Runtime-Support#3181] [Q&A] [answered]
Hi. I am a member of a small development company that has been doing business in Flash for nearly 25 years.
I've been trying to publish a Windows application using the latest AIR for the first time in several years, but I'm having trouble with signatures.
Currently, signing applications for Windows does not use `pfx` or `p12` files, and in fact, hardware tokens are now required.
However, Adobe Animate doesn't seem to be able to keep up with these changes. Since support has been discontinued, I think it will be difficult to improve.
What is the current best practice for signing? Do you want to build on Animate with a self-certificate and then re-sign it? (This hack became popular because the signature function was long broken in Animate on macOS)

While our team is good at animation and directing, there are many older people who are not good at programming, so we do the main development as much as possible on the Adobe Animate GUI, and many of us also edit the code on Animate.
I myself can handle CLI programs, but since most of the members are low-skilled, We would like to use Adobe Animate's GUI if possible for publishing application. I don't hope to write batch files or be the "build staff".

**Comment by ajwfrost** [comment, marked as answer]
Hi

For signing via Animate, you would have to contact the Adobe Animate team and raise this as a bug (or feature request? but a fairly crucial one as this will become more necessary as time goes by..). Adobe still officially supports Animate, but they do need some pressure to get them to make these kinds of updates.

I would suggest going with a self-signed certificate to create the package and then re-sign it afterwards... are you creating a 'bundle' i.e. a .exe file that needs to be signed via the Microsoft tools? Or if you're packaging as .air, you could use Animate to create .airi and then use `adt -sign` to create the .air file using a certificate based on a hardware token or similar.

thanks

**Comment by Kelsonk** [comment]
I'm still signing my Windows app in an ancient way using a pfx file. I'm not learning new things.
